<!DOCTYPE html>
<html lang="en-us">

<head>

  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="generator" content="Source Themes Academic 4.4.0">

  

  
  
  
  
  
    
    
    
  
  

  <meta name="author" content="Mark Goldberg">

  
  
  
    
  
  <meta name="description" content="Processing of sequencing data in R.">

  
  <link rel="alternate" hreflang="en-us" href="https://suvar.netlify.com/post/bioinformatics/hi-c/hi-c/">

  


  

  
  
  
  <meta name="theme-color" content="#2962ff">
  

  
  
  
  
    
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/academicons/1.8.6/css/academicons.min.css" integrity="sha256-uFVgMKfistnJAfoCUQigIl+JfUaP47GrRKjf6CTPVmw=" crossorigin="anonymous">
    <link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.6.0/css/all.css" integrity="sha384-aOkxzJ5uQz7WBObEZcHvV5JvRW3TUc2rNPA7pe3AwnsUohiw1Vj2Rgx2KSOkF5+h" crossorigin="anonymous">
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/fancybox/3.2.5/jquery.fancybox.min.css" integrity="sha256-ygkqlh3CYSUri3LhQxzdcm0n1EQvH2Y+U5S2idbLtxs=" crossorigin="anonymous">

    
    
    
      
    
    
      
      
        <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.15.6/styles/github.min.css" crossorigin="anonymous" title="hl-light">
        <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.15.6/styles/github.min.css" crossorigin="anonymous" title="hl-dark" disabled>
      
    

    

    

  

  
  
  <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Montserrat:400,700|Roboto:400,400italic,700|Roboto+Mono&display=swap">
  

  
  
  
  <link rel="stylesheet" href="/css/academic.min.3fab984c66eafc8a56bcfb5b5945e567.css">

  

  
  
    <script>
      window.ga=window.ga||function(){(ga.q=ga.q||[]).push(arguments)};ga.l=+new Date;
      ga('create', 'UA-144846434-1', 'auto');
      
      ga('require', 'eventTracker');
      ga('require', 'outboundLinkTracker');
      ga('require', 'urlChangeTracker');
      ga('send', 'pageview');
    </script>
    <script async src="https://www.google-analytics.com/analytics.js"></script>
    
    <script async src="https://cdnjs.cloudflare.com/ajax/libs/autotrack/2.4.1/autotrack.js" integrity="sha512-HUmooslVKj4m6OBu0OgzjXXr+QuFYy/k7eLI5jdeEy/F4RSgMn6XRWRGkFi5IFaFgy7uFTkegp3Z0XnJf3Jq+g==" crossorigin="anonymous"></script>
    
  
  

  

  

  <link rel="icon" type="image/png" href="/img/icon.png">
  <link rel="apple-touch-icon" type="image/png" href="/img/icon-192.png">

  <link rel="canonical" href="https://suvar.netlify.com/post/bioinformatics/hi-c/hi-c/">

  
  
  
  
    
  
  
  <meta property="twitter:card" content="summary_large_image">
  
  <meta property="og:site_name" content="Mark Goldberg">
  <meta property="og:url" content="https://suvar.netlify.com/post/bioinformatics/hi-c/hi-c/">
  <meta property="og:title" content="Processing of sequencing data in R | Mark Goldberg">
  <meta property="og:description" content="Processing of sequencing data in R."><meta property="og:image" content="https://suvar.netlify.com/img/logo.png">
  <meta property="twitter:image" content="https://suvar.netlify.com/img/logo.png"><meta property="og:locale" content="en-us">
  
    
      <meta property="article:published_time" content="2020-05-07T00:00:00&#43;00:00">
    
    <meta property="article:modified_time" content="2020-05-07T00:00:00&#43;00:00">
  

  


  





  <title>Processing of sequencing data in R | Mark Goldberg</title>

</head>

<body id="top" data-spy="scroll" data-offset="70" data-target="#TableOfContents" >

  <aside class="search-results" id="search">
  <div class="container">
    <section class="search-header">

      <div class="row no-gutters justify-content-between mb-3">
        <div class="col-6">
          <h1>Search</h1>
        </div>
        <div class="col-6 col-search-close">
          <a class="js-search" href="#"><i class="fas fa-times-circle text-muted" aria-hidden="true"></i></a>
        </div>
      </div>

      <div id="search-box">
        
        <input name="q" id="search-query" placeholder="Search..." autocapitalize="off"
        autocomplete="off" autocorrect="off" spellcheck="false" type="search">
        
      </div>

    </section>
    <section class="section-search-results">

      <div id="search-hits">
        
      </div>

    </section>
  </div>
</aside>


  
<nav class="navbar navbar-light fixed-top navbar-expand-lg py-0 compensate-for-scrollbar" id="navbar-main">
  <div class="container">

    
      <a class="navbar-brand" href="/">Mark Goldberg</a>
      
      <button type="button" class="navbar-toggler" data-toggle="collapse"
              data-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
        <span><i class="fas fa-bars"></i></span>
      </button>
      

    
    <div class="collapse navbar-collapse" id="navbar">

      
      
      <ul class="navbar-nav ml-auto">
        

        

        
        
        
          
        

        
        
        
        
        
        
          
          
          
            
          
          
        

        <li class="nav-item">
          <a class="nav-link " href="/#intro"><span>Home</span></a>
        </li>

        
        

        

        
        
        
          
        

        
        
        
        
        
        
          
          
          
            
          
          
        

        <li class="nav-item">
          <a class="nav-link " href="/#posts"><span>News</span></a>
        </li>

        
        

        

        
        
        

        
        
        
        
        
        
          
          
          
            
          
          
        

        <li class="nav-item">
          <a class="nav-link " href="/#R"><span>R</span></a>
        </li>

        
        

        

        
        
        
          
        

        
        
        
        
        
        

        <li class="nav-item">
          <a class="nav-link " href="/post/statistics/"><span>Math</span></a>
        </li>

        
        

        

        
        
        
          
        

        
        
        
        
        
        

        <li class="nav-item">
          <a class="nav-link " href="/post/biology/"><span>Biology</span></a>
        </li>

        
        

        

        
        
        
          
        

        
        
        
        
        
        

        <li class="nav-item">
          <a class="nav-link  active" href="/post/bioinformatics/"><span>Bioinformatics</span></a>
        </li>

        
        

        

        
        
        
          
        

        
        
        
        
        
        
          
          
          
            
          
          
        

        <li class="nav-item">
          <a class="nav-link " href="/#about"><span>About</span></a>
        </li>

        
        

        

        
        
        
          
        

        
        
        
        
        
        
          
          
          
            
          
          
        

        <li class="nav-item">
          <a class="nav-link " href="/#contact"><span>Contact</span></a>
        </li>

        
        

        
        <li class="nav-item dropdown">
          <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-haspopup="true"><span>Books</span><span class="caret"></span>
          </a>
          <ul class="dropdown-menu">
            
            <li class="dropdown-item my-0 py-0 mx-0 px-0">
              <a href="/echo.msk.ru"><span>Book 1</span></a>
            </li>
            
            <li class="dropdown-item my-0 py-0 mx-0 px-0">
              <a href="/google.com"><span>Book 2</span></a>
            </li>
            
          </ul>
        </li>

        
        

      

        

        
        <li class="nav-item">
          <a class="nav-link js-search" href="#"><i class="fas fa-search" aria-hidden="true"></i></a>
        </li>
        

        

        

      </ul>

    </div>
  </div>
</nav>


  <article class="article" itemscope itemtype="http://schema.org/Article">

  












  

  
  
  
<div class="article-container pt-3">
  <h1 itemprop="name">Processing of sequencing data in R</h1>

  

  
    



<meta content="2020-05-07 00:00:00 &#43;0000 UTC" itemprop="datePublished">
<meta content="2020-05-07 00:00:00 &#43;0000 UTC" itemprop="dateModified">

<div class="article-metadata">

  
  

  
  <span class="article-date">
    
    
      
    
    <time>May 7, 2020</time>
  </span>
  

  

  
  <span class="middot-divider"></span>
  <span class="article-reading-time">
    4 min read
  </span>
  

  
  
  
  <span class="middot-divider"></span>
  <a href="/post/bioinformatics/hi-c/hi-c/#disqus_thread"></a>
  

  
  
  <span class="middot-divider"></span>
  <span class="article-categories">
    <i class="fas fa-folder mr-1"></i><a href="/categories/bioinformatics/">Bioinformatics</a></span>
  

  
    
<div class="share-box" aria-hidden="true">
  <ul class="share">
    
      
      
      
        
      
      
      
      <li>
        <a href="https://twitter.com/intent/tweet?url=https://suvar.netlify.com/post/bioinformatics/hi-c/hi-c/&amp;text=Processing%20of%20sequencing%20data%20in%20R" target="_blank" rel="noopener" class="share-btn-twitter">
          <i class="fab fa-twitter"></i>
        </a>
      </li>
    
      
      
      
        
      
      
      
      <li>
        <a href="https://www.facebook.com/sharer.php?u=https://suvar.netlify.com/post/bioinformatics/hi-c/hi-c/&amp;t=Processing%20of%20sequencing%20data%20in%20R" target="_blank" rel="noopener" class="share-btn-facebook">
          <i class="fab fa-facebook-f"></i>
        </a>
      </li>
    
      
      
      
        
      
      
      
      <li>
        <a href="mailto:?subject=Processing%20of%20sequencing%20data%20in%20R&amp;body=https://suvar.netlify.com/post/bioinformatics/hi-c/hi-c/" target="_blank" rel="noopener" class="share-btn-email">
          <i class="fas fa-envelope"></i>
        </a>
      </li>
    
      
      
      
        
      
      
      
      <li>
        <a href="https://www.linkedin.com/shareArticle?url=https://suvar.netlify.com/post/bioinformatics/hi-c/hi-c/&amp;title=Processing%20of%20sequencing%20data%20in%20R" target="_blank" rel="noopener" class="share-btn-linkedin">
          <i class="fab fa-linkedin-in"></i>
        </a>
      </li>
    
      
      
      
        
      
      
      
      <li>
        <a href="https://web.whatsapp.com/send?text=Processing%20of%20sequencing%20data%20in%20R%20https://suvar.netlify.com/post/bioinformatics/hi-c/hi-c/" target="_blank" rel="noopener" class="share-btn-whatsapp">
          <i class="fab fa-whatsapp"></i>
        </a>
      </li>
    
      
      
      
        
      
      
      
      <li>
        <a href="https://service.weibo.com/share/share.php?url=https://suvar.netlify.com/post/bioinformatics/hi-c/hi-c/&amp;title=Processing%20of%20sequencing%20data%20in%20R" target="_blank" rel="noopener" class="share-btn-weibo">
          <i class="fab fa-weibo"></i>
        </a>
      </li>
    
  </ul>
</div>


  

</div>

    














  
</div>



  <div class="article-container">

    <div class="article-style" itemprop="articleBody">
      

<div id="TOC">
<ul>
<li><a href="#using-straw-tool-to-extract-contact-matrix-from-.hic-files">Using straw tool to extract contact matrix from .hic files</a></li>
<li><a href="#hic-pro-pipeline">HiC-Pro pipeline</a></li>
<li><a href="#multihic-compare-for-compare-of-2-hi-c-datasets-rao-et-al.2017">multiHiC compare for compare of 2 Hi-C datasets (Rao et al. 2017)</a></li>
<li><a href="#joint-normalization-of-hi-c">Joint normalization of Hi-C</a></li>
<li><a href="#difference-detection">Difference detection</a></li>
</ul>
</div>

<p>Workflow from Stansfield et al., Current Protocols in Bioinformatics, 2019</p>
<p>Hi-C workflow steps:<br />
1. mapping reads<br />
2. assigning fragments<br />
3. filtering fragments<br />
4. binning<br />
5. bin-level filtering<br />
6. balancing (normalization) of individual matrices</p>
<p>Paired-end reads of Hi-C experiments are mapped using the single-end mode to map each read (of the pair) independently.<br />
The theoretical maximum resolution of Hi-C sequencing is set by the restriction enzyme used to cut the DNA. However, most Hi-C datasets are not sequenced deeply enough to reach this theoretical maximum, and typically one of a few fixed-size resolutions are chosen for analyzing the data, including 1 Mb, 100 kb, 50 kb, 40 kb, 20 kb, 10 kb, and 5 kb.<br />
Two of the more popular pipelines for aligning Hi-C data are <strong>juicer</strong> (Durand, Shamim, &amp; Aiden, 2016) and <strong>HiC-Pro</strong> (Servant et al., 2015).</p>
<p><a href="https://github.com/aidenlab/juicer/wiki"><strong>juicer</strong></a> takes fastq files and aligns the data into .hic sparse contact maps. Alignment is based on BWA. Contact maps can be extracted from .hic files using juicer or the command line tool <strong>straw</strong>.</p>
<div id="using-straw-tool-to-extract-contact-matrix-from-.hic-files" class="section level2">
<h2>Using straw tool to extract contact matrix from .hic files</h2>
<pre class="bash"><code>&lt;NONE/VC/VC_SQRT/KR&gt; &lt;hicFile(s)&gt; &lt;chr1&gt;[:x1:x2] &lt;chr2&gt;[:y1:y2] &lt;BP/FRAG&gt; &lt;binsize&gt;</code></pre>
<p>VC - vanilla coverage normalization<br />
VC_SQRT - square root of vanilla coverage normalization<br />
KR - Knight-Ruiz (KR) normalization<br />
BP/FRAG - base pare or fragment size. Typically we use BP.</p>
<pre class="bash"><code>mkdir ~/GEO
cd ~/GEO
# download .hic file [9 Gb]
wget ftp://ftp.ncbi.nlm.nih.gov/geo/series/GSE63nnn/GSE63525/suppl/GSE63525_K562_combined_30.hic
# extract raw matrix for 22 chromosome at 500-kb resolution
straw NONE GSE63525_K562_combined_30.hic 22 22 BP 500000 &gt; K562.chHCT116_r22.500 kb.txt</code></pre>
<p>txt file is in the sparse upper-triangular matrix format, containing 3 columns:<br />
1. start of interaction<br />
2. end of interaction<br />
3. frequency (IF)</p>
</div>
<div id="hic-pro-pipeline" class="section level2">
<h2>HiC-Pro pipeline</h2>
<p>Output 2 files (.matrix and .bed):<br />
* The .matrix is plain-text 3-column sparse upper-triangular matrix with the columns <span class="math inline">\(bin_i\)</span> <span class="math inline">\(bin_j\)</span> and <span class="math inline">\(counts_ij\)</span>.<br />
* The .bed file contains the genomic coordinates for each <span class="math inline">\(bin_i\)</span> and <span class="math inline">\(bin_j\)</span>.</p>
<p><code>sparse2full</code> - convert sparse upper-triangular matrix to full contact matrix.<br />
<code>hicpro2bedpe</code> - convert alignments by HiC-Pro into BEDPE format.<br />
<code>hicpro2bedpe</code> - input .matrix and .bed and convert into sparse upper-triangular matrix.</p>
</div>
<div id="multihic-compare-for-compare-of-2-hi-c-datasets-rao-et-al.2017" class="section level2">
<h2>multiHiC compare for compare of 2 Hi-C datasets (Rao et al. 2017)</h2>
<pre class="bash"><code>mkdir ~/GEO
cd ~/GEO
# download .hic files from
wget ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM2795nnn/GSM2795535/suppl/GSM2795535_Rao-2017-HIC001_30.hic
wget ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM2795nnn/GSM2795536/suppl/GSM2795536_Rao-2017-HIC002_30.hic
wget ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM2809nnn/GSM2809539/suppl/GSM2809539_Rao-2017-HIC008_30.hic
wget ftp://ftp.ncbi.nlm.nih.gov/geo/samples/GSM2809nnn/GSM2809540/suppl/GSM2809540_Rao-2017-HIC009_30.hic

# convert .hic to sparse contact matrix for each of 22 chromosomes
for i in {1,2,8,9}
   do
     mkdir HIC00${i}
     for j in {1..22}
       do
         straw NONE *_Rao-2017-HIC00${i}_30.hic $j $j BP 100000 &gt; HIC00${i}/HIC00${i}.NONE.chr${j}.100000.txt
       done
   done</code></pre>
<pre class="r"><code>library(readr)
library(data.table)
library(dplyr)
library(edgeR)
library(BiocParallel)
library(HiCcompare)
library(multiHiCcompare)


options(scipen = 10) # output fixed numbers

# Set up parameters for reading in data
chr &lt;- paste0(&#39;chr&#39;, c(1:22))     # Chromosome names
samples &lt;- paste0(&#39;HIC00&#39;, c(1,2,8,9)) # Sample names
res &lt;- 100000                          # Data resolution

# Read data
sample_list &lt;- list()
chr_list &lt;- list()
wd &lt;- &#39;/home/suvar/GEO/&#39;
for(j in 1:length(samples)) {
  for (i in 1:length(chr)) {
    chr_list[[i]] &lt;- read_tsv(paste0(wd, samples[j], &#39;/&#39;,
    samples[j], &#39;.NONE.&#39;, chr[i], &#39;.&#39;, res, &#39;.txt&#39;),
    col_names = FALSE) %&gt;% as.data.table()
    # Add column indicating the chromosome
    chr_list[[i]] &lt;- cbind(i, chr_list[[i]])
    colnames(chr_list[[i]]) &lt;- c(&#39;chr&#39;, &#39;region1&#39;, &#39;region2&#39;, &#39;IF&#39;)
  }
  sample_list[[j]] &lt;- chr_list
  chr_list &lt;- list()
}

# Collapse separate chromosome lists into one table per sample
sample_list &lt;- lapply(sample_list, rbindlist)
  
sample_list[[1]]</code></pre>
</div>
<div id="joint-normalization-of-hi-c" class="section level1">
<h1>Joint normalization of Hi-C</h1>
<pre class="r"><code>library(pander)
# Create a Hicexp object for use by multiHiCcompare (~10 min)
# Four objects are assigned into two groups
rao2017 &lt;- make_hicexp(data_list = sample_list, groups = c(1,1,2,2))
rao2017 # class(rao2017)

# view the IF information
hic_table(rao2017)

# MD plots before normalization
MD_hicexp(rao2017, plot.chr = 1, plot.loess = TRUE)</code></pre>
<pre class="r"><code># Normalize (~2 min)
rao2017 &lt;- fastlo(rao2017) # cyclic loesss normalisation is also available

# Plot normalization results
MD_hicexp(rao2017, plot.chr = 1, plot.loess = TRUE)

# Print normalized IFs
pander::pandoc.table(head(hic_table(rao2017)))

library(BiocParallel)
# Check how many cores are available
numCores &lt;- parallel::detectCores()
# Set the number of cores at least one less than the total number

if(Sys.info()[&#39;sysname&#39;] == &#39;Windows&#39;) {
  # Windows settings
  register(SnowParam(workers = numCores-1),
  default = TRUE)
  }else {
  # Unix settings
  register(MulticoreParam(workers = numCores-1),
  default = TRUE)
  }</code></pre>
</div>
<div id="difference-detection" class="section level1">
<h1>Difference detection</h1>
<pre class="r"><code># Perform exact test (~10 min)
# May use &quot;parallel = TRUE&quot; option to speed up computations
rao2017 &lt;- hic_exactTest(rao2017, parallel = TRUE)
# Plot a composite MD plot with the results of a comparison
MD_composite(rao2017,plot.chr = 1)
# Print results as a data frame
pander::pandoc.table(head(results(rao2017)))

# Save the Hicexp object
save(rao2017, file = paste0(wd,&#39;rao2017.RDA&#39;))
# To start the downstream analysis
# without re-running multiHiCcompare load the saved file
# load(paste0(wd,&#39;rao2017.RDA&#39;))</code></pre>
</div>

    </div>

    


    

<div class="article-tags">
  
  <a class="badge badge-light" href="/tags/r/">R</a>
  
  <a class="badge badge-light" href="/tags/rna-seq/">RNA-seq</a>
  
</div>



    
      








  





  
  
  
    
  
  
  <div class="media author-card" itemscope itemtype="http://schema.org/Person">
    
      
      <img class="portrait mr-3" src="/authors/admin/avatar_hubf35e96ec2f6be37cf8c4f36be371ec7_39026_250x250_fill_q90_lanczos_center.jpg" itemprop="image" alt="Avatar">
    

    <div class="media-body">
      <h5 class="card-title" itemprop="name"><a href="https://suvar.netlify.com/">Mark Goldberg</a></h5>
      <h6 class="card-subtitle">Researcher</h6>
      <p class="card-text" itemprop="description">My research interests include epigenetics and computational biology.</p>
      <ul class="network-icon" aria-hidden="true">
        
          
          
          
            
          
          
          
          
          
            
          
          <li>
            <a itemprop="sameAs" href="/#contact" >
              <i class="fas fa-envelope"></i>
            </a>
          </li>
        
          
          
          
            
          
          
          
          
          
            
          
          <li>
            <a itemprop="sameAs" href="https://twitter.com/suvarlab" target="_blank" rel="noopener">
              <i class="fab fa-twitter"></i>
            </a>
          </li>
        
          
          
          
            
          
          
          
          
          
            
          
          <li>
            <a itemprop="sameAs" href="https://telegram.me/@suvarlab" target="_blank" rel="noopener">
              <i class="fab fa-telegram"></i>
            </a>
          </li>
        
      </ul>
    </div>
  </div>



      
      
      <div class="article-widget">
        <div class="hr-light"></div>
        <h3>Related</h3>
        <ul>
          
          <li><a href="/post/bioinformatics/processing_sequencing_data_in_r/rna_seq_analysis_in_r/">Processing of sequencing data in R</a></li>
          
          <li><a href="/post/bioinformatics/rna_seq_in_r/rna_seq_analysis_in_r/">RNA-seq analysis in R</a></li>
          
          <li><a href="/post/r/dplyr_tutorial/dplyr_tutorial/">Dplyr - reference</a></li>
          
          <li><a href="/post/statistics/model_reference/models_reference/">Models - reference</a></li>
          
          <li><a href="/post/r/r_graphic/r_graphic/">R graphic - reference</a></li>
          
        </ul>
      </div>
      
    

    

    
<section id="comments">
  
    
<div id="disqus_thread"></div>
<script>
  let disqus_config = function () {
    
    
    
  };
  (function() {
    if (["localhost", "127.0.0.1"].indexOf(window.location.hostname) != -1) {
      document.getElementById('disqus_thread').innerHTML = 'Disqus comments not available by default when the website is previewed locally.';
      return;
    }
    var d = document, s = d.createElement('script'); s.async = true;
    s.src = 'https://' + "suvar-1" + '.disqus.com/embed.js';
    s.setAttribute('data-timestamp', +new Date());
    (d.head || d.body).appendChild(s);
  })();
</script>
<noscript>Please enable JavaScript to view the <a href="https://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
<a href="https://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>


  
</section>



  </div>
</article>

      

    
    
    
    <script src="/js/mathjax-config.js"></script>
    

    
    
    
      <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
      <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery.imagesloaded/4.1.4/imagesloaded.pkgd.min.js" integrity="sha256-lqvxZrPLtfffUl2G/e7szqSvPBILGbwmsGE1MKlOi0Q=" crossorigin="anonymous"></script>
      <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery.isotope/3.0.6/isotope.pkgd.min.js" integrity="sha256-CBrpuqrMhXwcLLUd5tvQ4euBHCdh7wGlDfNz8vbu/iI=" crossorigin="anonymous"></script>
      <script src="https://cdnjs.cloudflare.com/ajax/libs/fancybox/3.2.5/jquery.fancybox.min.js" integrity="sha256-X5PoE3KU5l+JcX+w09p/wHl9AzK333C4hJ2I9S5mD4M=" crossorigin="anonymous"></script>

      
        <script src="https://cdnjs.cloudflare.com/ajax/libs/mermaid/8.0.0/mermaid.min.js" integrity="sha256-0w92bcB21IY5+rGI84MGj52jNfHNbXVeQLrZ0CGdjNY=" crossorigin="anonymous" title="mermaid"></script>
      

      
        
        <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.15.6/highlight.min.js" integrity="sha256-aYTdUrn6Ow1DDgh5JTc3aDGnnju48y/1c8s1dgkYPQ8=" crossorigin="anonymous"></script>
        
        <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.15.6/languages/r.min.js"></script>
        
      

      
      
      <script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.4/MathJax.js?config=TeX-AMS_CHTML-full" integrity="sha256-GhM+5JHb6QUzOQPXSJLEWP7R73CbkisjzK5Eyij4U9w=" crossorigin="anonymous" async></script>
      
    

    
    

    
    
    <script>hljs.initHighlightingOnLoad();</script>
    

    
    
    <script>
      const search_index_filename = "/index.json";
      const i18n = {
        'placeholder': "Search...",
        'results': "results found",
        'no_results': "No results found"
      };
      const content_type = {
        'post': "Posts",
        'project': "Projects",
        'publication' : "Publications",
        'talk' : "Talks"
        };
    </script>
    

    
    

    
    
    <script id="search-hit-fuse-template" type="text/x-template">
      <div class="search-hit" id="summary-{{key}}">
      <div class="search-hit-content">
        <div class="search-hit-name">
          <a href="{{relpermalink}}">{{title}}</a>
          <div class="article-metadata search-hit-type">{{type}}</div>
          <p class="search-hit-description">{{snippet}}</p>
        </div>
      </div>
      </div>
    </script>
    

    
    
    <script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/3.2.1/fuse.min.js" integrity="sha256-VzgmKYmhsGNNN4Ph1kMW+BjoYJM2jV5i4IlFoeZA9XI=" crossorigin="anonymous"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/jquery.mark.min.js" integrity="sha256-4HLtjeVgH0eIB3aZ9mLYF6E8oU5chNdjU6p6rrXpl9U=" crossorigin="anonymous"></script>
    

    
    

    
    
    
    <script id="dsq-count-scr" src="https://suvar-1.disqus.com/count.js" async></script>
    

    
    
    
    
    
    
    
    
    
      
    
    
    
    
    <script src="/js/academic.min.b22d2266dab94c8508ae1f0cf6b7056c.js"></script>

    






  
  <div class="container">
    <footer class="site-footer">
  

  <p class="powered-by">
    © 2019 &middot; 

    Powered by the
    <a href="https://sourcethemes.com/academic/" target="_blank" rel="noopener">Academic theme</a> for
    <a href="https://gohugo.io" target="_blank" rel="noopener">Hugo</a>.

    
    <span class="float-right" aria-hidden="true">
      <a href="#" id="back_to_top">
        <span class="button_icon">
          <i class="fas fa-chevron-up fa-2x"></i>
        </span>
      </a>
    </span>
    
  </p>
</footer>

  </div>
  

  
<div id="modal" class="modal fade" role="dialog">
  <div class="modal-dialog">
    <div class="modal-content">
      <div class="modal-header">
        <h5 class="modal-title">Cite</h5>
        <button type="button" class="close" data-dismiss="modal" aria-label="Close">
          <span aria-hidden="true">&times;</span>
        </button>
      </div>
      <div class="modal-body">
        <pre><code class="tex hljs"></code></pre>
      </div>
      <div class="modal-footer">
        <a class="btn btn-outline-primary my-1 js-copy-cite" href="#" target="_blank">
          <i class="fas fa-copy"></i> Copy
        </a>
        <a class="btn btn-outline-primary my-1 js-download-cite" href="#" target="_blank">
          <i class="fas fa-download"></i> Download
        </a>
        <div id="modal-error"></div>
      </div>
    </div>
  </div>
</div>

</body>
</html>
