<!DOCTYPE html>
<html lang="en-us">

<head>

  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1">
  <meta http-equiv="X-UA-Compatible" content="IE=edge">
  <meta name="generator" content="Source Themes Academic 4.4.0">

  

  
  
  
  
  
    
    
    
  
  

  <meta name="author" content="Mark Goldberg">

  
  
  
    
  
  <meta name="description" content="Here you can find a beautiful example of differential expression analysis using standard R packages.">

  
  <link rel="alternate" hreflang="en-us" href="https://suvar.netlify.com/post/bioinformatics/differential_expression_analysis/diferential_expression_analysis/">

  


  

  
  
  
  <meta name="theme-color" content="#2962ff">
  

  
  
  
  
    
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/academicons/1.8.6/css/academicons.min.css" integrity="sha256-uFVgMKfistnJAfoCUQigIl+JfUaP47GrRKjf6CTPVmw=" crossorigin="anonymous">
    <link rel="stylesheet" href="https://use.fontawesome.com/releases/v5.6.0/css/all.css" integrity="sha384-aOkxzJ5uQz7WBObEZcHvV5JvRW3TUc2rNPA7pe3AwnsUohiw1Vj2Rgx2KSOkF5+h" crossorigin="anonymous">
    <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/fancybox/3.2.5/jquery.fancybox.min.css" integrity="sha256-ygkqlh3CYSUri3LhQxzdcm0n1EQvH2Y+U5S2idbLtxs=" crossorigin="anonymous">

    
    
    
      
    
    
      
      
        <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.15.6/styles/github.min.css" crossorigin="anonymous" title="hl-light">
        <link rel="stylesheet" href="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.15.6/styles/github.min.css" crossorigin="anonymous" title="hl-dark" disabled>
      
    

    

    

  

  
  
  <link rel="stylesheet" href="https://fonts.googleapis.com/css?family=Montserrat:400,700|Roboto:400,400italic,700|Roboto+Mono&display=swap">
  

  
  
  
  <link rel="stylesheet" href="/css/academic.min.3fab984c66eafc8a56bcfb5b5945e567.css">

  

  
  
    <script>
      window.ga=window.ga||function(){(ga.q=ga.q||[]).push(arguments)};ga.l=+new Date;
      ga('create', 'UA-144846434-1', 'auto');
      
      ga('require', 'eventTracker');
      ga('require', 'outboundLinkTracker');
      ga('require', 'urlChangeTracker');
      ga('send', 'pageview');
    </script>
    <script async src="https://www.google-analytics.com/analytics.js"></script>
    
    <script async src="https://cdnjs.cloudflare.com/ajax/libs/autotrack/2.4.1/autotrack.js" integrity="sha512-HUmooslVKj4m6OBu0OgzjXXr+QuFYy/k7eLI5jdeEy/F4RSgMn6XRWRGkFi5IFaFgy7uFTkegp3Z0XnJf3Jq+g==" crossorigin="anonymous"></script>
    
  
  

  

  

  <link rel="icon" type="image/png" href="/img/icon.png">
  <link rel="apple-touch-icon" type="image/png" href="/img/icon-192.png">

  <link rel="canonical" href="https://suvar.netlify.com/post/bioinformatics/differential_expression_analysis/diferential_expression_analysis/">

  
  
  
  
    
  
  
  <meta property="twitter:card" content="summary_large_image">
  
  <meta property="og:site_name" content="Mark Goldberg">
  <meta property="og:url" content="https://suvar.netlify.com/post/bioinformatics/differential_expression_analysis/diferential_expression_analysis/">
  <meta property="og:title" content="Differential expression analysis | Mark Goldberg">
  <meta property="og:description" content="Here you can find a beautiful example of differential expression analysis using standard R packages."><meta property="og:image" content="https://suvar.netlify.com/img/logo.png">
  <meta property="twitter:image" content="https://suvar.netlify.com/img/logo.png"><meta property="og:locale" content="en-us">
  
    
      <meta property="article:published_time" content="2019-08-09T00:00:00&#43;00:00">
    
    <meta property="article:modified_time" content="2019-08-09T00:00:00&#43;00:00">
  

  


  





  <title>Differential expression analysis | Mark Goldberg</title>

</head>

<body id="top" data-spy="scroll" data-offset="70" data-target="#TableOfContents" >

  <aside class="search-results" id="search">
  <div class="container">
    <section class="search-header">

      <div class="row no-gutters justify-content-between mb-3">
        <div class="col-6">
          <h1>Search</h1>
        </div>
        <div class="col-6 col-search-close">
          <a class="js-search" href="#"><i class="fas fa-times-circle text-muted" aria-hidden="true"></i></a>
        </div>
      </div>

      <div id="search-box">
        
        <input name="q" id="search-query" placeholder="Search..." autocapitalize="off"
        autocomplete="off" autocorrect="off" spellcheck="false" type="search">
        
      </div>

    </section>
    <section class="section-search-results">

      <div id="search-hits">
        
      </div>

    </section>
  </div>
</aside>


  
<nav class="navbar navbar-light fixed-top navbar-expand-lg py-0 compensate-for-scrollbar" id="navbar-main">
  <div class="container">

    
      <a class="navbar-brand" href="/">Mark Goldberg</a>
      
      <button type="button" class="navbar-toggler" data-toggle="collapse"
              data-target="#navbar" aria-controls="navbar" aria-expanded="false" aria-label="Toggle navigation">
        <span><i class="fas fa-bars"></i></span>
      </button>
      

    
    <div class="collapse navbar-collapse" id="navbar">

      
      
      <ul class="navbar-nav ml-auto">
        

        

        
        
        
          
        

        
        
        
        
        
        
          
          
          
            
          
          
        

        <li class="nav-item">
          <a class="nav-link " href="/#intro"><span>Home</span></a>
        </li>

        
        

        

        
        
        
          
        

        
        
        
        
        
        
          
          
          
            
          
          
        

        <li class="nav-item">
          <a class="nav-link " href="/#posts"><span>News</span></a>
        </li>

        
        

        

        
        
        

        
        
        
        
        
        
          
          
          
            
          
          
        

        <li class="nav-item">
          <a class="nav-link " href="/#R"><span>R</span></a>
        </li>

        
        

        

        
        
        
          
        

        
        
        
        
        
        

        <li class="nav-item">
          <a class="nav-link " href="/post/statistics/"><span>Math</span></a>
        </li>

        
        

        

        
        
        
          
        

        
        
        
        
        
        

        <li class="nav-item">
          <a class="nav-link " href="/post/biology/"><span>Biology</span></a>
        </li>

        
        

        

        
        
        
          
        

        
        
        
        
        
        

        <li class="nav-item">
          <a class="nav-link  active" href="/post/bioinformatics/"><span>Bioinformatics</span></a>
        </li>

        
        

        

        
        
        
          
        

        
        
        
        
        
        
          
          
          
            
          
          
        

        <li class="nav-item">
          <a class="nav-link " href="/#about"><span>About</span></a>
        </li>

        
        

        

        
        
        
          
        

        
        
        
        
        
        
          
          
          
            
          
          
        

        <li class="nav-item">
          <a class="nav-link " href="/#contact"><span>Contact</span></a>
        </li>

        
        

        
        <li class="nav-item dropdown">
          <a href="#" class="nav-link dropdown-toggle" data-toggle="dropdown" aria-haspopup="true"><span>Books</span><span class="caret"></span>
          </a>
          <ul class="dropdown-menu">
            
            <li class="dropdown-item my-0 py-0 mx-0 px-0">
              <a href="https://suvarzz.github.io/statbook"><span>Statistics</span></a>
            </li>
            
            <li class="dropdown-item my-0 py-0 mx-0 px-0">
              <a href="https://suvarzz.github.io/rbook/"><span>Rbook</span></a>
            </li>
            
          </ul>
        </li>

        
        

      

        

        
        <li class="nav-item">
          <a class="nav-link js-search" href="#"><i class="fas fa-search" aria-hidden="true"></i></a>
        </li>
        

        

        

      </ul>

    </div>
  </div>
</nav>


  <article class="article" itemscope itemtype="http://schema.org/Article">

  












  

  
  
  
<div class="article-container pt-3">
  <h1 itemprop="name">Differential expression analysis</h1>

  

  
    



<meta content="2019-08-09 00:00:00 &#43;0000 UTC" itemprop="datePublished">
<meta content="2019-08-09 00:00:00 &#43;0000 UTC" itemprop="dateModified">

<div class="article-metadata">

  
  

  
  <span class="article-date">
    
    
      
    
    <time>Aug 9, 2019</time>
  </span>
  

  

  
  <span class="middot-divider"></span>
  <span class="article-reading-time">
    6 min read
  </span>
  

  
  
  
  <span class="middot-divider"></span>
  <a href="/post/bioinformatics/differential_expression_analysis/diferential_expression_analysis/#disqus_thread"></a>
  

  
  
  <span class="middot-divider"></span>
  <span class="article-categories">
    <i class="fas fa-folder mr-1"></i><a href="/categories/statistics/">Statistics</a></span>
  

  
    
<div class="share-box" aria-hidden="true">
  <ul class="share">
    
      
      
      
        
      
      
      
      <li>
        <a href="https://twitter.com/intent/tweet?url=https://suvar.netlify.com/post/bioinformatics/differential_expression_analysis/diferential_expression_analysis/&amp;text=Differential%20expression%20analysis" target="_blank" rel="noopener" class="share-btn-twitter">
          <i class="fab fa-twitter"></i>
        </a>
      </li>
    
      
      
      
        
      
      
      
      <li>
        <a href="https://www.facebook.com/sharer.php?u=https://suvar.netlify.com/post/bioinformatics/differential_expression_analysis/diferential_expression_analysis/&amp;t=Differential%20expression%20analysis" target="_blank" rel="noopener" class="share-btn-facebook">
          <i class="fab fa-facebook-f"></i>
        </a>
      </li>
    
      
      
      
        
      
      
      
      <li>
        <a href="mailto:?subject=Differential%20expression%20analysis&amp;body=https://suvar.netlify.com/post/bioinformatics/differential_expression_analysis/diferential_expression_analysis/" target="_blank" rel="noopener" class="share-btn-email">
          <i class="fas fa-envelope"></i>
        </a>
      </li>
    
      
      
      
        
      
      
      
      <li>
        <a href="https://www.linkedin.com/shareArticle?url=https://suvar.netlify.com/post/bioinformatics/differential_expression_analysis/diferential_expression_analysis/&amp;title=Differential%20expression%20analysis" target="_blank" rel="noopener" class="share-btn-linkedin">
          <i class="fab fa-linkedin-in"></i>
        </a>
      </li>
    
      
      
      
        
      
      
      
      <li>
        <a href="https://web.whatsapp.com/send?text=Differential%20expression%20analysis%20https://suvar.netlify.com/post/bioinformatics/differential_expression_analysis/diferential_expression_analysis/" target="_blank" rel="noopener" class="share-btn-whatsapp">
          <i class="fab fa-whatsapp"></i>
        </a>
      </li>
    
      
      
      
        
      
      
      
      <li>
        <a href="https://service.weibo.com/share/share.php?url=https://suvar.netlify.com/post/bioinformatics/differential_expression_analysis/diferential_expression_analysis/&amp;title=Differential%20expression%20analysis" target="_blank" rel="noopener" class="share-btn-weibo">
          <i class="fab fa-weibo"></i>
        </a>
      </li>
    
  </ul>
</div>


  

</div>

    














  
</div>



  <div class="article-container">

    <div class="article-style" itemprop="articleBody">
      

<div id="TOC">
<ul>
<li><a href="#gene-expression-analysis-of-histone-deacetylase-1-hdac1-knockout-mouse.">Gene expression analysis of histone deacetylase 1 (HDAC1) knockout mouse.</a></li>
<li><a href="#sources">Sources</a></li>
</ul>
</div>

<div id="gene-expression-analysis-of-histone-deacetylase-1-hdac1-knockout-mouse." class="section level2">
<h2>Gene expression analysis of histone deacetylase 1 (HDAC1) knockout mouse.</h2>
<p>This short tutorial should help to understand the basic principal of gene expression analysis using simple dataset and nearly basic R.</p>
<ul>
<li>Affymetrix microarray</li>
<li><em>Dataset</em>: <a href="http://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE5583">GSE5583</a><br />
</li>
<li><em>Paper</em>: Mol Cell Biol 2006 Nov;26(21):7913-28. <a href="http://www.ncbi.nlm.nih.gov/pubmed/16940178">PMID: 16940178</a><br />
</li>
<li><em>R code</em>: Ahmed Moustafa</li>
</ul>
<pre class="r"><code># Read the data into R
library (RCurl)
url = getURL (&quot;http://bit.ly/GSE5583_data&quot;, followlocation = TRUE)
data = as.matrix(read.table (text = url, row.names = 1, header = T))

# Check the loaded dataset
dim(data) # Dimension of the dataset</code></pre>
<pre><code>## [1] 12488     6</code></pre>
<pre class="r"><code># data shows gene experssion levels in 6 samples:
# rows correspond to samples (3 wild type WT and 3 knock-out KO)
# columns correspond to genes ids
head(data) # First few rows</code></pre>
<pre><code>##           WT.GSM130365 WT.GSM130366 WT.GSM130367 KO.GSM130368 KO.GSM130369
## 100001_at         11.5          5.6         69.1         15.7         36.0
## 100002_at         20.5         32.4         93.3         31.8         14.4
## 100003_at         72.4         89.0         79.2         80.5        130.1
## 100004_at        261.0        226.2        365.1        432.0        447.3
## 100005_at       1086.2       1555.6       1487.1       1062.2       1365.9
## 100006_at         49.7         52.9         15.0         25.8         48.8
##           KO.GSM130370
## 100001_at         42.0
## 100002_at         22.9
## 100003_at         86.7
## 100004_at        288.1
## 100005_at       1436.2
## 100006_at         54.8</code></pre>
<pre class="r"><code>###################
# Exploratory plots
###################

# Check the behavior of the data
hist(data, col = &quot;gray&quot;, main=&quot;GSE5583 - Histogram&quot;)</code></pre>
<p><img src="/post/bioinformatics/differential_expression_analysis/diferential_expression_analysis_files/figure-html/unnamed-chunk-1-1.png" width="672" /></p>
<pre class="r"><code># Log2 transformation (why?)
data2 = log2(data)

# Check the behavior of the data after log-transformation
hist(data2, col = &quot;gray&quot;, main=&quot;GSE5583 (log2) - Histogram&quot;)</code></pre>
<p><img src="/post/bioinformatics/differential_expression_analysis/diferential_expression_analysis_files/figure-html/unnamed-chunk-1-2.png" width="672" /></p>
<pre class="r"><code># Boxplot
boxplot(data2, col=c(&quot;darkgreen&quot;, &quot;darkgreen&quot;, &quot;darkgreen&quot;,
                     &quot;darkred&quot;, &quot;darkred&quot;, &quot;darkred&quot;),
        main=&quot;GSE5583 - boxplots&quot;, las=2)</code></pre>
<p><img src="/post/bioinformatics/differential_expression_analysis/diferential_expression_analysis_files/figure-html/unnamed-chunk-1-3.png" width="672" /></p>
<pre class="r"><code># Hierarchical clustering of the &quot;samples&quot; based on
# the correlation coefficients of the expression values
hc = hclust(as.dist(1-cor(data2)))
plot(hc, main=&quot;GSE5583 - Hierarchical Clustering&quot;)</code></pre>
<p><img src="/post/bioinformatics/differential_expression_analysis/diferential_expression_analysis_files/figure-html/unnamed-chunk-1-4.png" width="672" /></p>
<pre class="r"><code>#######################################
# Differential expression (DE) analysis
#######################################

# Separate the two conditions into two smaller data frames
wt = data2[,1:3]
ko = data2[,4:6]

# Compute the means of the samples of each condition
wt.mean = apply(wt, 1, mean)
ko.mean = apply(ko, 1, mean)

head(wt.mean)</code></pre>
<pre><code>## 100001_at 100002_at 100003_at 100004_at 100005_at 100006_at 
##  4.039868  5.306426  6.320360  8.120503 10.408872  5.089087</code></pre>
<pre class="r"><code>head(ko.mean)</code></pre>
<pre><code>## 100001_at 100002_at 100003_at 100004_at 100005_at 100006_at 
##  4.844978  4.452076  6.597451  8.576804 10.318839  5.358071</code></pre>
<pre class="r"><code># Just get the maximum of all the means
limit = max(wt.mean, ko.mean)

# Scatter plot
plot(ko.mean ~ wt.mean, xlab = &quot;WT&quot;, ylab = &quot;KO&quot;,
     main = &quot;GSE5583 - Scatter&quot;, xlim = c(0, limit), ylim = c(0, limit))
# Diagonal line
abline(0, 1, col = &quot;red&quot;)</code></pre>
<p><img src="/post/bioinformatics/differential_expression_analysis/diferential_expression_analysis_files/figure-html/unnamed-chunk-1-5.png" width="672" /></p>
<pre class="r"><code># Compute fold-change (biological significance)
# Difference between the means of the conditions
fold = wt.mean - ko.mean

# Histogram of the fold differences
hist(fold, col = &quot;gray&quot;)</code></pre>
<p><img src="/post/bioinformatics/differential_expression_analysis/diferential_expression_analysis_files/figure-html/unnamed-chunk-1-6.png" width="672" /></p>
<pre class="r"><code># Compute statistical significance (using t-test)
pvalue = NULL # Empty list for the p-values
tstat = NULL # Empty list of the t test statistics

for(i in 1 : nrow(data)) { # For each gene : 
  x = wt[i,] # WT of gene number i
  y = ko[i,] # KO of gene number i
  
  # Compute t-test between the two conditions
  t = t.test(x, y)
  
  # Put the current p-value in the pvalues list
  pvalue[i] = t$p.value
  # Put the current t-statistic in the tstats list
  tstat[i] = t$statistic
}

head(pvalue)</code></pre>
<pre><code>## [1] 0.5449730 0.3253745 0.3287830 0.1892376 0.6928410 0.7180077</code></pre>
<pre class="r"><code># Histogram of p-values (-log10)
hist(-log10(pvalue), col = &quot;gray&quot;)</code></pre>
<p><img src="/post/bioinformatics/differential_expression_analysis/diferential_expression_analysis_files/figure-html/unnamed-chunk-1-7.png" width="672" /></p>
<pre class="r"><code># Volcano: put the biological significance (fold-change)
# and statistical significance (p-value) in one plot
plot(fold, -log10(pvalue), main = &quot;GSE5583 - Volcano&quot;)

fold_cutoff = 2
pvalue_cutoff = 0.01
abline(v = fold_cutoff, col = &quot;blue&quot;, lwd = 3)
abline(v = -fold_cutoff, col = &quot;red&quot;, lwd = 3)
abline(h = -log10(pvalue_cutoff), col = &quot;green&quot;, lwd = 3)</code></pre>
<p><img src="/post/bioinformatics/differential_expression_analysis/diferential_expression_analysis_files/figure-html/unnamed-chunk-1-8.png" width="672" /></p>
<pre class="r"><code># Screen for the genes that satisfy the filtering criteria

# Fold-change filter for &quot;biological&quot; significance
filter_by_fold = abs(fold) &gt;= fold_cutoff
dim(data2[filter_by_fold, ])</code></pre>
<pre><code>## [1] 210   6</code></pre>
<pre class="r"><code># P-value filter for &quot;statistical&quot; significance
filter_by_pvalue = pvalue &lt;= pvalue_cutoff
dim(data2[filter_by_pvalue, ])</code></pre>
<pre><code>## [1] 429   6</code></pre>
<pre class="r"><code># Combined filter (both biological and statistical)
filter_combined = filter_by_fold &amp; filter_by_pvalue

filtered = data2[filter_combined,]
dim(filtered)</code></pre>
<pre><code>## [1] 42  6</code></pre>
<pre class="r"><code>head(filtered)</code></pre>
<pre><code>##             WT.GSM130365 WT.GSM130366 WT.GSM130367 KO.GSM130368
## 100716_at       4.852998     4.906891     5.626439     7.572890
## 100914_at      10.340852     9.917074    10.250062    12.248787
## 101368_at       9.937227    10.204693    10.385215    12.270354
## 101550_at       5.526695     5.439623     6.221104     2.137504
## 101635_f_at     7.105385     6.722466     6.943687     5.266787
## 101883_s_at     5.768184     6.127221     5.133399    11.564292
##             KO.GSM130369 KO.GSM130370
## 100716_at       7.791163     7.299208
## 100914_at      12.185526    12.127124
## 101368_at      12.213499    12.078184
## 101550_at       2.906891     2.035624
## 101635_f_at     4.842979     4.643856
## 101883_s_at    11.679568    11.663514</code></pre>
<pre class="r"><code># Let&#39;s generate the volcano plot again,
# highlighting the significantly differential expressed genes
plot(fold, -log10(pvalue), main = &quot;GSE5583 - Volcano #2&quot;)
points (fold[filter_combined], -log10(pvalue[filter_combined]),
        pch = 16, col = &quot;red&quot;)</code></pre>
<p><img src="/post/bioinformatics/differential_expression_analysis/diferential_expression_analysis_files/figure-html/unnamed-chunk-1-9.png" width="672" /></p>
<pre class="r"><code># Highlighting up-regulated in red and down-regulated in blue
plot(fold, -log10(pvalue), main = &quot;GSE5583 - Volcano #3&quot;)
points (fold[filter_combined &amp; fold &lt; 0],
        -log10(pvalue[filter_combined &amp; fold &lt; 0]),
        pch = 16, col = &quot;red&quot;)
points (fold[filter_combined &amp; fold &gt; 0],
        -log10(pvalue[filter_combined &amp; fold &gt; 0]),
        pch = 16, col = &quot;blue&quot;)</code></pre>
<p><img src="/post/bioinformatics/differential_expression_analysis/diferential_expression_analysis_files/figure-html/unnamed-chunk-1-10.png" width="672" /></p>
<pre class="r"><code># Cluster the rows (genes) &amp; columns (samples) by correlation
rowv = as.dendrogram(hclust(as.dist(1-cor(t(filtered)))))
colv = as.dendrogram(hclust(as.dist(1-cor(filtered))))

# Generate a heatmap
heatmap(filtered, Rowv=rowv, Colv=colv, cexCol=0.7)</code></pre>
<p><img src="/post/bioinformatics/differential_expression_analysis/diferential_expression_analysis_files/figure-html/unnamed-chunk-1-11.png" width="672" /></p>
<pre class="r"><code>library(gplots)

# Enhanced heatmap
heatmap.2(filtered, Rowv=rowv, Colv=colv, cexCol=0.7,
          col = rev(redblue(256)), scale = &quot;row&quot;,
          trace=&quot;none&quot;, density.info=&quot;none&quot;)</code></pre>
<p><img src="/post/bioinformatics/differential_expression_analysis/diferential_expression_analysis_files/figure-html/unnamed-chunk-2-1.png" width="672" /></p>
<pre class="r"><code># Save the heatmap to a PDF file
pdf (&quot;GSE5583_DE_Heatmap.pdf&quot;)
heatmap.2(filtered, Rowv=rowv, Colv=colv, cexCol=0.7,
          col = rev(redblue(256)), scale = &quot;row&quot;)
dev.off()

# Save the DE genes to a text file
write.table (filtered, &quot;GSE5583_DE.txt&quot;, sep = &quot;\t&quot;,
             quote = FALSE)</code></pre>
<pre class="r"><code>n = nrow(filtered)

cor.table = NULL
x = NULL
y = NULL
cor.val = NULL
cor.sig = NULL

for (i in 1 : (n-1)) {
  x_name = rownames(filtered)[i]
  x_exps = filtered[i, ]    
  
  for (j in (i+1) : n) {
    y_name = rownames(filtered)[j]
    y_exps = filtered[j, ]
    
    output = cor.test(x_exps,y_exps)
    
    x = c(x, x_name)
    y = c(y, y_name)
    cor.val = c(cor.val, output$estimate)
    cor.sig = c(cor.sig, output$p.value)
  }
}

cor.table = data.frame (x, y, cor.val, cor.sig)

dim(cor.table)</code></pre>
<pre><code>## [1] 861   4</code></pre>
<pre class="r"><code>head(cor.table)</code></pre>
<pre><code>##           x           y    cor.val      cor.sig
## 1 100716_at   100914_at  0.9732295 0.0010653980
## 2 100716_at   101368_at  0.9897688 0.0001564799
## 3 100716_at   101550_at -0.9060431 0.0128271221
## 4 100716_at 101635_f_at -0.9433403 0.0047245418
## 5 100716_at 101883_s_at  0.9508680 0.0035616301
## 6 100716_at   102712_at  0.9676037 0.0015572795</code></pre>
<pre class="r"><code>sig_cutoff = 0.001

cor.filtered = subset (cor.table, cor.sig &lt; sig_cutoff)

dim(cor.filtered)</code></pre>
<pre><code>## [1] 314   4</code></pre>
<pre class="r"><code>head(cor.filtered)</code></pre>
<pre><code>##            x         y    cor.val      cor.sig
## 2  100716_at 101368_at  0.9897688 1.564799e-04
## 8  100716_at 103088_at -0.9761495 8.464861e-04
## 10 100716_at 103299_at -0.9991089 1.190632e-06
## 14 100716_at 104700_at -0.9792543 6.411095e-04
## 15 100716_at 160172_at  0.9833552 4.132702e-04
## 16 100716_at 160943_at  0.9814703 5.118449e-04</code></pre>
</div>
<div id="sources" class="section level2">
<h2>Sources</h2>
<p><a href="https://gist.github.com/ahmedmoustafa">Ahmed Moustafa githab</a></p>
</div>

    </div>

    


    

<div class="article-tags">
  
  <a class="badge badge-light" href="/tags/r/">R</a>
  
  <a class="badge badge-light" href="/tags/statistics/">Statistics</a>
  
  <a class="badge badge-light" href="/tags/clustering/">Clustering</a>
  
  <a class="badge badge-light" href="/tags/hierarchical/">Hierarchical</a>
  
</div>



    
      








  





  
  
  
    
  
  
  <div class="media author-card" itemscope itemtype="http://schema.org/Person">
    
      
      <img class="portrait mr-3" src="/authors/admin/avatar_hubf35e96ec2f6be37cf8c4f36be371ec7_39026_250x250_fill_q90_lanczos_center.jpg" itemprop="image" alt="Avatar">
    

    <div class="media-body">
      <h5 class="card-title" itemprop="name"><a href="https://suvar.netlify.com/">Mark Goldberg</a></h5>
      <h6 class="card-subtitle">Researcher</h6>
      <p class="card-text" itemprop="description">My research interests include epigenetics and computational biology.</p>
      <ul class="network-icon" aria-hidden="true">
        
          
          
          
            
          
          
          
          
          
            
          
          <li>
            <a itemprop="sameAs" href="/#contact" >
              <i class="fas fa-envelope"></i>
            </a>
          </li>
        
          
          
          
            
          
          
          
          
          
            
          
          <li>
            <a itemprop="sameAs" href="https://twitter.com/suvarlab" target="_blank" rel="noopener">
              <i class="fab fa-twitter"></i>
            </a>
          </li>
        
          
          
          
            
          
          
          
          
          
            
          
          <li>
            <a itemprop="sameAs" href="https://telegram.me/@suvarlab" target="_blank" rel="noopener">
              <i class="fab fa-telegram"></i>
            </a>
          </li>
        
      </ul>
    </div>
  </div>



      
      
      <div class="article-widget">
        <div class="hr-light"></div>
        <h3>Related</h3>
        <ul>
          
          <li><a href="/post/statistics/hierarchical_cluster/hierarchical_cluser/">Hierarchical cluster analysis</a></li>
          
          <li><a href="/post/statistics/k_means/k_means/">k-means</a></li>
          
          <li><a href="/post/statistics/determin_number_clusters/determin_number_clusters/">Determining the optimal number of clusters</a></li>
          
          <li><a href="/post/statistics/exploration/exploration/">Exploratory analysis in R</a></li>
          
          <li><a href="/post/statistics/compare_models_caret/compare_models_caret/">Compare Models And Select The Best Using The Caret R Package</a></li>
          
        </ul>
      </div>
      
    

    

    
<section id="comments">
  
    
<div id="disqus_thread"></div>
<script>
  let disqus_config = function () {
    
    
    
  };
  (function() {
    if (["localhost", "127.0.0.1"].indexOf(window.location.hostname) != -1) {
      document.getElementById('disqus_thread').innerHTML = 'Disqus comments not available by default when the website is previewed locally.';
      return;
    }
    var d = document, s = d.createElement('script'); s.async = true;
    s.src = 'https://' + "suvar-1" + '.disqus.com/embed.js';
    s.setAttribute('data-timestamp', +new Date());
    (d.head || d.body).appendChild(s);
  })();
</script>
<noscript>Please enable JavaScript to view the <a href="https://disqus.com/?ref_noscript">comments powered by Disqus.</a></noscript>
<a href="https://disqus.com" class="dsq-brlink">comments powered by <span class="logo-disqus">Disqus</span></a>


  
</section>



  </div>
</article>

      

    
    
    
    <script src="/js/mathjax-config.js"></script>
    

    
    
    
      <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery/3.4.1/jquery.min.js" integrity="sha256-CSXorXvZcTkaix6Yvo6HppcZGetbYMGWSFlBw8HfCJo=" crossorigin="anonymous"></script>
      <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery.imagesloaded/4.1.4/imagesloaded.pkgd.min.js" integrity="sha256-lqvxZrPLtfffUl2G/e7szqSvPBILGbwmsGE1MKlOi0Q=" crossorigin="anonymous"></script>
      <script src="https://cdnjs.cloudflare.com/ajax/libs/jquery.isotope/3.0.6/isotope.pkgd.min.js" integrity="sha256-CBrpuqrMhXwcLLUd5tvQ4euBHCdh7wGlDfNz8vbu/iI=" crossorigin="anonymous"></script>
      <script src="https://cdnjs.cloudflare.com/ajax/libs/fancybox/3.2.5/jquery.fancybox.min.js" integrity="sha256-X5PoE3KU5l+JcX+w09p/wHl9AzK333C4hJ2I9S5mD4M=" crossorigin="anonymous"></script>

      
        <script src="https://cdnjs.cloudflare.com/ajax/libs/mermaid/8.0.0/mermaid.min.js" integrity="sha256-0w92bcB21IY5+rGI84MGj52jNfHNbXVeQLrZ0CGdjNY=" crossorigin="anonymous" title="mermaid"></script>
      

      
        
        <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.15.6/highlight.min.js" integrity="sha256-aYTdUrn6Ow1DDgh5JTc3aDGnnju48y/1c8s1dgkYPQ8=" crossorigin="anonymous"></script>
        
        <script src="https://cdnjs.cloudflare.com/ajax/libs/highlight.js/9.15.6/languages/r.min.js"></script>
        
      

      
      
      <script src="https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.4/MathJax.js?config=TeX-AMS_CHTML-full" integrity="sha256-GhM+5JHb6QUzOQPXSJLEWP7R73CbkisjzK5Eyij4U9w=" crossorigin="anonymous" async></script>
      
    

    
    

    
    
    <script>hljs.initHighlightingOnLoad();</script>
    

    
    
    <script>
      const search_index_filename = "/index.json";
      const i18n = {
        'placeholder': "Search...",
        'results': "results found",
        'no_results': "No results found"
      };
      const content_type = {
        'post': "Posts",
        'project': "Projects",
        'publication' : "Publications",
        'talk' : "Talks"
        };
    </script>
    

    
    

    
    
    <script id="search-hit-fuse-template" type="text/x-template">
      <div class="search-hit" id="summary-{{key}}">
      <div class="search-hit-content">
        <div class="search-hit-name">
          <a href="{{relpermalink}}">{{title}}</a>
          <div class="article-metadata search-hit-type">{{type}}</div>
          <p class="search-hit-description">{{snippet}}</p>
        </div>
      </div>
      </div>
    </script>
    

    
    
    <script src="https://cdnjs.cloudflare.com/ajax/libs/fuse.js/3.2.1/fuse.min.js" integrity="sha256-VzgmKYmhsGNNN4Ph1kMW+BjoYJM2jV5i4IlFoeZA9XI=" crossorigin="anonymous"></script>
    <script src="https://cdnjs.cloudflare.com/ajax/libs/mark.js/8.11.1/jquery.mark.min.js" integrity="sha256-4HLtjeVgH0eIB3aZ9mLYF6E8oU5chNdjU6p6rrXpl9U=" crossorigin="anonymous"></script>
    

    
    

    
    
    
    <script id="dsq-count-scr" src="https://suvar-1.disqus.com/count.js" async></script>
    

    
    
    
    
    
    
    
    
    
      
    
    
    
    
    <script src="/js/academic.min.b22d2266dab94c8508ae1f0cf6b7056c.js"></script>

    






  
  <div class="container">
    <footer class="site-footer">
  

  <p class="powered-by">
    © 2019 &middot; 

    Powered by the
    <a href="https://sourcethemes.com/academic/" target="_blank" rel="noopener">Academic theme</a> for
    <a href="https://gohugo.io" target="_blank" rel="noopener">Hugo</a>.

    
    <span class="float-right" aria-hidden="true">
      <a href="#" id="back_to_top">
        <span class="button_icon">
          <i class="fas fa-chevron-up fa-2x"></i>
        </span>
      </a>
    </span>
    
  </p>
</footer>

  </div>
  

  
<div id="modal" class="modal fade" role="dialog">
  <div class="modal-dialog">
    <div class="modal-content">
      <div class="modal-header">
        <h5 class="modal-title">Cite</h5>
        <button type="button" class="close" data-dismiss="modal" aria-label="Close">
          <span aria-hidden="true">&times;</span>
        </button>
      </div>
      <div class="modal-body">
        <pre><code class="tex hljs"></code></pre>
      </div>
      <div class="modal-footer">
        <a class="btn btn-outline-primary my-1 js-copy-cite" href="#" target="_blank">
          <i class="fas fa-copy"></i> Copy
        </a>
        <a class="btn btn-outline-primary my-1 js-download-cite" href="#" target="_blank">
          <i class="fas fa-download"></i> Download
        </a>
        <div id="modal-error"></div>
      </div>
    </div>
  </div>
</div>

</body>
</html>
